src/compiler/crystal/tools/doc/markdown/parser.cr

```diff
@@ -115,6 +116,12 @@ class Crystal::Doc::Markdown::Parser
       .strip                     # Strip leading/trailing whitespace
       .gsub(/[\s_-]+/, '-')      # Replace `_` and leftover whitespace with `-`
 
+    @anchor_map[anchor] += 1
+
+    if (seen_count = @anchor_map[anchor]) && (seen_count > 1)
```

Comment:
`(seen_count = @anchor_map[anchor])` doesn't make sense as a condition. It's always truthy. It never even raises because of the previous line.

Reply:
Good point, I thought of a simpler way to handle it.